R/lazyframe__lazy.R

```diff
@@ -270,6 +270,27 @@ LazyFrame_collect_background = function() {
   .pr$LazyFrame$collect_background(self)
 }
 
+#' @title Collect_in_background
+#' @description collect DataFrame by lazy query in a background thread.
+#' @details
+#' Useful to not block the R session while query executes. Any use of `$map()` or `$apply()` is
+#' diverted to other R sessions. If R functions rely on global variables this can change the
+#' result.
+#'
+#' @keywords LazyFrame DataFrame_new
+#' @return a thread handle for the task
+#' @examples
+#' prexpr <- pl$col("mpg")$map(\(x) {Sys.sleep(1.5);x * 0.43}, in_background = TRUE)$alias("kml")
+#' handle = pl$LazyFrame(mtcars)$with_column(prexpr)$collect_in_background()
+#' if(!handle$is_finished()) print("not done yet")
+#' df = handle$join() #get result
+#' df
+#' handle$is_finished()
+#' handle$join()
```

Comment:
it seems the informative text from thiserror does not come through to the user
```R
Error: Execution halted with the following contexts
0: During function call [handle$join()]
1: Handled
```
It should state `Joined on a used thread handler`. A casual R user may not understand what just `Handled` means.

Reply:
@Sicheng-Pan I have resolved this now, use of Rctx::handled().into() was problematic because Rctx is casted to dyn Error and then back to Rctx.plain().